src/Couchbase.Lite.Shared/API/Error/CouchbaseLiteErrorMessage.cs

```diff
@@ -0,0 +1,84 @@
+using System.Text;
+
+namespace Couchbase.Lite
+{
+    internal static class CouchbaseLiteErrorMessage
+    {
+        //Database error messages
+        internal const string CreateDBDirectoryFailed = "Unable to create database directory";
+        //Database - Copy
+        internal const string ResolveDefaultDirectoryFailed = "Failed to resolve a default directory! If you have overriden the IDefaultDirectoryResolver interface, please check it.  Otherwise please file a bug report.";
+        internal const string InvalidPath = "Path.Combine failed to return a non-null value!";
+        //Database - Close
+        internal const string CloseDBFailedReplications = "Cannot close the database. Please stop all of the replicators before closing the database.";
+        internal const string CloseDBFailedQueryListeners = "Cannot close the database. Please remove all of the query listeners before closing the database.";
+        //Database - Delete (Save with deletion =  true)
+        internal const string DeleteDBFailedReplications = "Cannot delete the database. Please stop all of the replicators before closing the database.";
+        internal const string DeleteDBFailedQueryListeners = "Cannot delete the database. Please remove all of the query listeners before closing the database.";
+        internal const string DeleteDocFailedNotSaved = "Cannot delete a document that has not yet been saved";
+        //Database - Purge
+        internal const string DocumentNotFound = "Document doesn't exist in the database.";
+        internal const string DocumentAnotherDatabase = "Cannot operate on a document from another database";
+        //Database - Save
+        internal const string BlobDifferentDatabase = "A document contains a blob that was saved to a different database; the save operation cannot complete.";
+        internal const string BlobContentNull = "No data available to write for install. Please ensure that all blobs in the document have non-null content.";
+        //Database - Resolve Conflict
+        internal const string ResolvedDocContainsNull = "Resolved document contains a null body";
+        internal const string ResolvedDocFailedLiteCore = "LiteCore failed resolving conflict.";
+        internal const string ResolvedDocWrongDb = "Resolved document db {0} is different from expected db {1}";
+        internal const string DBClosed = "Attempt to perform an operation on a closed database";
+        internal const string NoDocumentRevision = "No revision data on the document!";
+        //Fragment - Value
+        internal const string FragmentPathNotExist = "Specified fragment path does not exist in object, cannot set value";
+        //ToCouchbaseObject(object value))
+        internal const string InvalidCouchbaseObjType = "{0} is not a valid type. " +
+                                                "You may only pass byte, sbyte, short, ushort, int, uint, long, ulong, float, double, bool, DateTimeOffset, Blob, " +
```

Comment:
in case of iOS, `Data, Int, Int64, Float, Double, String, Date, Bool, Blob, Any`

how about `You may only pass language supported types`?

Reply:
Too vague.  Maybe this needs to be another formatted message.